api_server.py

```
@@ -75,98 +84,94 @@ def format_hours(minutes: Optional[int]) -> Optional[str]:
-        duration_min = int((end_time - now).total_seconds() / 60)
-
-        display = end_time.strftime("%H:%M")
-        if end_time.date() == now.date():
-            display += " today"
-        elif end_time.date() == (now + timedelta(days=1)).date():
-            display += " tomorrow"
-        else:
-            display += end_time.strftime(" on %d/%m")
+    conn = get_db()
+    curr = conn.execute(
+        f"SELECT end_time FROM {table} WHERE {col} = ? AND start_time <= ? AND end_time > ? LIMIT 1",
```

Ruff (0.15.5)

[error] Possible SQL injection vector through string-based query construction

(S608)